synapse/storage/receipts.py

```diff
@@ -407,7 +407,6 @@ def insert_linearized_receipt_txn(self, txn, room_id, receipt_type,
                 txn,
                 room_id=room_id,
                 user_id=user_id,
-                topological_ordering=topological_ordering,
```

Comment:
wonder if we ought to make the comparison at line 375 ignore topo ordering. Might fix the fact I currently have 1110 unread messages in #matrix-dev too.